libhb/muxavformat.c

```diff
@@ -624,26 +623,40 @@ static int avformatInit( hb_mux_object_t * m )
     }
 
     hb_rational_t vrate = job->vrate;
+    hb_rational_t clock_vrate = { clock, av_rescale(vrate.den, clock, vrate.num)};
+    int standard_rate = 0;
 
-    // If the vrate is the internal clock rate, there's a good chance
-    // this is a standard rate that we have in our hb_video_rates table.
+    // Check if the vrate is similar to a standard rate that we have in our hb_video_rates table.
     // Because of rounding errors and approximations made while
     // measuring framerate, the actual value may not be exact.  So
-    // we look for rates that are "close" and make an adjustment
-    // to fps.den.
-    if (vrate.num == clock)
+    // we look for rates that are "close"
+    const hb_rate_t *video_framerate = NULL;
+    while ((video_framerate = hb_video_framerate_get_next(video_framerate)) != NULL)
     {
-        const hb_rate_t *video_framerate = NULL;
-        while ((video_framerate = hb_video_framerate_get_next(video_framerate)) != NULL)
+        if (abs(clock_vrate.den - video_framerate->rate) < 10)
         {
-            if (abs(vrate.den - video_framerate->rate) < 10)
-            {
-                vrate.den = video_framerate->rate;
-                break;
-            }
+            vrate.num = clock;
+            vrate.den = video_framerate->rate;
+            standard_rate = 1;
+            break;
         }
     }
+
     hb_reduce(&vrate.num, &vrate.den, vrate.num, vrate.den);
+
+    if (job->mux == HB_MUX_AV_MP4 && standard_rate &&
+        job->cfr == 1 && vrate.den * 90000 % vrate.num)
```

Comment:
I believe it's possible to receive a job->vrate that didn't reduce well which could cause an overflow in `vrate.den * 90000`. I suggest making this `vrate.den * 90000L` to force a long conversion to be on the safe side.

Aside from that, looks good to me.

Reply:
Done, thanks for the review ☺️